### ISSUE TITLE
Additional Sensor Data from Console Output

### ISSUE THREAD

**Issue by muzzymate** [nzapponi/homebridge-simplisafe3#11]
I am assuming that this is what you mean when you asked to provide sample data. My apologies if I missed instructions elsewhere, but am happy to assist. Please let me know if there is any additional information you need. I blanked out the serial numbers but can message you privately if you require them. 

Motion and Glassbreak sensors information is not being surfaced in the console, but it looks like you've already determined that those do return a status value. 

**Panic Button -** 

[9/4/2019, 2:36:25 PM] [Home Alarm] Sensor not (yet) supported: Bedroom
[9/4/2019, 2:36:25 PM] [Home Alarm] {
  serial: 'XXXXXXXX',
  type: 3,
  status: {},
  name: 'Bedroom',
  setting: { alarm: 1 },
  flags: { swingerShutdown: false, lowBattery: false, offline: false }
}

**Smoke Detector -** 

[9/4/2019, 2:36:25 PM] [Home Alarm] Sensor not (yet) supported: Staircase
[9/4/2019, 2:36:25 PM] [Home Alarm] {
  serial: 'XXXXXXXX',
  type: 8,
  status: { test: false, tamper: false, malfunction: false, triggered: false },
  name: 'Staircase',
  setting: {},
  flags: { swingerShutdown: false, lowBattery: false, offline: false }
}

**Water Sensor -** 

[9/4/2019, 2:36:25 PM] [Home Alarm] Sensor not (yet) supported: Sump Pump
[9/4/2019, 2:36:25 PM] [Home Alarm] {
  serial: 'XXXXXXXX',
  type: 9,
  status: { triggered: false },
  name: 'Sump Pump',
  setting: { alarm: 1 },
  flags: { swingerShutdown: false, lowBattery: false, offline: false }
}

**Temperature Sensor -** 

[9/4/2019, 2:36:25 PM] [Home Alarm] Sensor not (yet) supported: Basement
[9/4/2019, 2:36:25 PM] [Home Alarm] {
  serial: 'XXXXXXXX',
  type: 10,
  status: { temperature: 67, triggered: false },
  name: 'Basement',
  setting: { highTemp: 100, lowTemp: 41, alarm: 1 },
  flags: { swingerShutdown: false, lowBattery: false, offline: false }
}

**Keypad -**

[9/4/2019, 2:36:25 PM] [Home Alarm] Sensor not (yet) supported: Home
[9/4/2019, 2:36:25 PM] [Home Alarm] {
  serial: 'XXXXXXXX',
  type: 13,
  status: {},
  name: 'Home',
  setting: { alarmVolume: 3, doorChime: 0, exitBeeps: 0, entryBeeps: 0 },
  flags: { swingerShutdown: false, lowBattery: false, offline: false }
}

**Comment by graanco** [comment]
Hey buddy... I see your here too now... I am looking at the code and I am not sure if this one is better. There seems to be a lot of hops which might slow it down. I was hoping that this code used the web socket but I am unsure at this time if it really does. I see that there is a set interval telling me that it is still scrubbing the webpage as a pull.... I would like to improve this so it pushes the data verse pulling the sensor data. That way a lot less traffic to the internet connections. I am waiting for a response from them.

**Comment by muzzymate** [comment]
Yes, migrated to this plugin when you updated that you were "moving on" and pointed to this plugin. Trying it out now.

**Comment by graanco** [comment]
I've been updating the code the my old project once I get it going I think... It will work on all version of simple safe and allow multiple systems to be added.

Things I don't like seeing on this one is the plugins. I don't like them my only issue is the socket.io. I think I will need to use that one.

**Comment by nzapponi** [comment]
Thanks for providing the data!

Keypad and panic button do not report any status information (see `status: {}`), however smoke detector, temperature sensor and water sensor do, which is great news!

I'll add support for these this week.

Since I don't have those sensors myself, would you be happy to help me out by testing it when the update is ready?

Thanks!

**Comment by nzapponi** [comment]
@muzzymate could you please update the plugin and let me know if your smoke detector, water sensor and freeze sensor are now working?

**Comment by muzzymate** [comment]
Hi @nzapponi - The plugin updated last night and now I see the temperature, smoke, and water sensors appearing in the Home app. Temperature reading in Home matches what I see in the SimpliSafe app. I'll have to figure out a way to test the water and smoke sensors without triggering the calvary. 

One last thing, I do have a Glassbreak sensor joined to my system (can see in the SimpliSafe app). However, there is no information logged out to the console about it. Your updated readme file implies that you still need sample data for it.

**Comment by graanco** [comment]
Based on SSAPI 
https://webapp.simplisafe.com/c951c28ef02b_res/components/ss-control-panel/1.32.0/ss-control-panel.js


These are all the sensors

{unknown:-1,baseStation:0,keypad:1,keychainRemote:2,panicButton:3,motionSensor:4,entrySensor:5,glassbreakSensor:6,coDetector:7,smokeDetector:8,waterSensor:9,freezeSensor:10,siren:11,camera:1e3,nest:2e3}

Based on my code...

  /*Commented out sensors not used by Homebridge and yes I know the siren could be a speaker*/
      0:'SecuritySystem',
      /*1:'keypad',
      2:'keychain',
      3:'panic_button',*/
      4:'MotionSensor',
      5:'ContactSensor',
      6:'GlassBreakSensor', /* used as motion sensor */
      7:'CarbonMonoxideSensor',
      8:'SmokeSensor',
      9:'LeakSensor',
      10:'TemperatureSensor',
      /*11:'siren',*/
      1e3:'Camera',/*,
      -1:'unknown',*/

      'SecuritySystem': 0,
      /*'keypad': 1,
      'keychain': 2,
      'panic_button': 3,*/
      'MotionSensor': 4,
      'ContactSensor': 5,
      'GlassBreakSensor': 6, /* used as motion sensor */
      'CarbonMonoxideSensor': 7,
      'SmokeSensor': 8,
      'LeakSensor': 9,
      'TemperatureSensor': 10,
      /*'siren': 11,*/
      'Camera': 1e3/*,
      'unknown': -1*/
    };

Now I did find the base64 for the last image for the cameras but I am unable to find that again...

**Comment by nzapponi** [comment]
> One last thing, I do have a Glassbreak sensor joined to my system (can see in the SimpliSafe app). However, there is no information logged out to the console about it. Your updated readme file implies that you still need sample data for it.

@muzzymate I don't think that's necessary 😄 , as long as the devices appear correctly inside the Home app, you'll be fine! The way the status is reported is the same as an entry sensor.

@graanco thanks for that, I wasn't aware of that spec in the angular source code. I don't believe the glass break sensor actually reports any status though, right? 

I have just added a debug mode in v1.2.1, could you please update, add `"debug": true` to your `config.json`, and add here what data you get about your glassbreak sensor?

Thanks!

**Comment by muzzymate** [comment]
Here is the output of the Glassbreak Sensor. As you expected, no status information, sadly. 


[9/10/2019, 12:17:33 PM] [Home Alarm] Discovered sensor: Office
[9/10/2019, 12:17:33 PM] [Home Alarm] {
  serial: 'XXXXXXXX',
  type: 6,
  status: {},
  name: 'Office',
  setting: {
    instantTrigger: false,
    away2: 1,
    away: 1,
    home2: 1,
    home: 1,
    off: 0
  },
  flags: { swingerShutdown: false, lowBattery: false, offline: false }
}

**Comment by graanco** [comment]
First I wonder on how to use this is NODEJS...

https://cdn.jsdelivr.net/npm/socket.io-file-client@2.0.2/socket.io-file-client.min.js

Now my question are you still scrapping the website for updates? Or are you pursuing the web socket.io.... I still feel that having SS website update you the status on the system is a way better approach them consistently calling them.

About the glass break sensor... When I was scrapping the JSON I treated it the same as motion sensor...

case ss.SensorTypes.GlassBreakSensor:
case ss.SensorTypes.MotionSensor:
                    accessory.getService(Service.MotionSensor).getCharacteristic(Characteristic.MotionDetected).updateValue(sensors[sensor].status.triggered);
                    break;

Is how I wrote it in my code... All it needs to do is just indicate a trip notification from the  status.triggered.

**Comment by graanco** [comment]
In the SSAPI... I see that they treat these sensor as intrusionSensors

   case W.SensorType.Entry:
   case W.SensorType.Motion:
   case W.SensorType.GlassBreak:
    t = W.intrusionSensors;
    break;

That being stated... I think they all have the same trip flags.

**Comment by nzapponi** [comment]
@graanco yeah but where most sensors report a `status`, the ones I didn't include don't seem to.
For example, my motion sensor never reports `"status": { "triggered": true }`, even when I'm jumping in front of it! Makes me think that they're building a common API framework for all their sensors, but that not all their current ones report on their status. And until they do, I don't see much point of including them in the Home app just to show the same status all the time.

If you do have one of those other sensors, could you try and trigger them and see what happens?

In terms of socket.io and the likes, I have implemented socket.io for all the events that SS3 actually reports (arm / disarm / etc, see here: https://github.com/nzapponi/homebridge-simplisafe3/blob/97d537e7cdc4ccfd27188feeb4705f129c919ebd/src/simplisafe.js#L420 and here: https://github.com/nzapponi/homebridge-simplisafe3/blob/97d537e7cdc4ccfd27188feeb4705f129c919ebd/src/accessories/alarm.js#L143 ), however when I last looked at it, I wasn't receiving any sensor-specific events.

I saw that the angular source does contain a spec of events, but again I haven't received any... Have you?

This is the spec I found:
```
alarmSmokeDetectorTriggered: "1110",
    alarmWaterSensorTriggered: "1154",
    alarmFreezeSensorTriggered: "1159",
    alarmCoSensorTriggered: "1162",
    alarmEntrySensorTriggered: "1134",
    alarmMotionOrGlassbreakSensorTriggered: "1132",
    alarmPanicButtonTriggered: "1120",
    alarmCanceled: "1406",
    alarmSmokeDetectorStopped: "3110",
    alarmWaterSensorStopped: "3154",
    alarmFreezeSensorStopped: "3159",
    alarmCoSensorStopped: "3162",
    systemPowerOutage: "1301",
    systemPowerRestored: "3301",
    systemInterferenceDetected: "1344",
    systemInterferenceResolved: "3344",
    sensorError: "1381",
    sensorRestored: "3381",
    systemArmed: "3400",
    systemArmedHome: "3441",
    systemArmedAway: "3401",
    systemDisarmed: "1400",
    alertSecret: "1409",
    userRecording: "1609",
    cameraRecording: "1170",
    doorbellRang: "1458",
    testSignalReceivedUser: "1601",
    testSignalReceivedSensor: "1604",
    testSignalReceivedAuto: "1602",
    alarmOther: "1140",
    alarmHeatSensorTriggered: "1158",
    alarmHeatSensorStopped: "3158",
    medicalAlarm: "1100",
    systemAwayRemote: "3407",
    systemHome2: "3491",
    systemAway2: "3481",
    systemAway2Remote: "3487",
    batteryLow: "1302",
    batteryRestored: "3302",
    wiFiOutage: "1350",
    wiFiRestored: "3350",
    sensorPaired: "1531",
    otaDownloaded: "1416",
    entryDelay: "1429",
    warningSensorOpen: "1426",
    systemOff: "1407",
    systemHomeCount: "9441",
    systemAwayCount: "9401",
    systemAwayCountRemote: "9407",
    sensorAdded: "1531",
    sensorNamed: "1533",
    wiFiUpdateSuccess: "3360",
    wiFiUpdateFailure: "1360"
```

**Comment by graanco** [comment]
Unfortunately my current house I am at is using v2 and the v3 is in another state so I am unable to trigger that system. 

I used to get updates for v2 on the websocket fro the SS website. I have tried recently.

**Comment by nzapponi** [comment]
Ok, I’m not home today but I will test it again and see if I get any useful events through!

I’d rather use websockets than continuous requests if possible.

**Comment by graanco** [comment]
I agree about the sockets.. and don't forget about the cfgs

https://webapp.simplisafe.com/ssAppConfig.js

This ai wale to scrap and use

**Comment by graanco** [comment]
Quick question... I see this alarmEntrySensorTriggered and is that only for an alarm to trigger or does it report the status on the sensor... Like when the system is off the contact sensor is opened or close. Would it report that too?

**Comment by nzapponi** [comment]
Yeah not sure - I’ll test it this week. But I suspect it won’t report anything if the alarm is off, which means we’ll probably still need to poll continuously.
